api/admin/opds.py

```diff
@@ -66,7 +67,22 @@ def annotate_work_entry(self, work, active_license_pool, edition, identifier, fe
                 identifier_type=identifier.type,
                 identifier=identifier.identifier, _external=True)
         )
-            
+
+        # If there is a storage integration, changing the cover is allowed.
+        _db = Session.object_session(identifier)
+        storage_integrations = _db.query(ExternalIntegration).filter(
+            ExternalIntegration.goal==ExternalIntegration.STORAGE_GOAL
+        )
+        if storage_integrations.count() > 0:
+            feed.add_link_to_entry(
+                entry,
+                rel="http://librarysimplified.org/terms/rel/change_cover",
+                href=self.url_for(
+                    "work_change_book_cover",
+                    identifier_type=identifier.type,
+                    identifier=identifier.identifier, _external=True)
+            )
```

Comment:
The presence of this link is used to determine whether to show the "Cover" tab. If there's no storage integration, it won't show.

Reply:
This should only happen if there's a storage integration _for this work_, i.e. either a sitewide integration or an integration for one of its collections.

Refactoring https://github.com/NYPL-Simplified/circulation/pull/968/files#diff-d3633d1a0ad08a784fffc12ecdba4065R1185 might help this.

Not a big deal -- we could even simplify it by saying only a sitewide integration counts.